### PR TITLE
uv runs on mxmmake

### PR DESCRIPTION
**Pull request by jensens**
- [x] use `uv pip` via `PYTHON_PACKAGE_INSTALLER?=uv`
- [x] use `uv venv` via `PYTHON_UV_GLOBAL?=true`
- [x] fix tests

### CODE REVIEW

**Comment by jensens** [comment]
ready

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,20 @@ PRIMARY_PYTHON?=python3
 # Default: 3.7
 PYTHON_MIN_VERSION?=3.7
 
+# Install packages using the given package installer method.
+# Supported are `pip` and `uv`. If uv is used, its global availability is
+# checked. Otherwise, is is installed, either in the virtual environment or
```
Comment (thet) [inline Makefile:58]:
```suggestion
# checked. Otherwise, it is installed, either in the virtual environment or
```

File: Makefile [hunk 2]
```diff
@@ -208,18 +222,43 @@ else
...
 $(MXENV_TARGET): $(SENTINEL)
 ifeq ("$(VENV_ENABLED)", "true")
+# Use a venv
 ifeq ("$(VENV_CREATE)", "true")
-	@echo "Setup Python Virtual Environment under '$(VENV_FOLDER)'"
-	@$(PRIMARY_PYTHON) -m venv $(VENV_FOLDER)
-endif
-endif
+# Create a venv
+ifeq ("$(PYTHON_PACKAGE_INSTALLER)$(PYTHON_UV_GLOBAL)", "uvtrue")
```
Comment (thet) [inline Makefile:240]: :+1:

**Review by thet — APPROVED** [review]
LGTM

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -4,22 +4,26 @@ on: [push]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-
     strategy:
       fail-fast: false
       matrix:
-        python:
-        - "3.7"
```
Comment (rnixx) [inline .github/workflows/test.yml:13]: What's the problem with running 3.7?

File: Makefile [hunk 4]
```diff
@@ -53,6 +53,20 @@ PRIMARY_PYTHON?=python3
...
+# using the `PRIMARY_PYTHON`, dependent on the `VENV_ENABLED` setting. If
+# `VENV_ENABLED` and uv is selected, uv is used to create the virtual
+# environment.
+# Default: pip
+PYTHON_PACKAGE_INSTALLER?=uv
+
+# Flag whether to use a global installed 'uv' or install
+# it in the virtual environment.
+# Default: false
+PYTHON_UV_GLOBAL?=false
```
Comment (rnixx) [inline Makefile:68]: Is PYTHON the proper prefix? Should we name it just UV_GLOBAL or maybe MXENV_UV_GLOBAL after the domain name?

Comment (jensens) [reply Makefile:68]: I would like to be explicit here, since uv is a short name and I have no idea where it may conflict in future.

Comment (jensens) [reply Makefile:68]: MXENV_UV_GLOBAL would be OK too. given mxenv is Python related only

**Pull request merged**